### ISSUE TITLE
scale_pos_weight in custom_objective

### ISSUE THREAD

**Issue by yzliao** [dmlc/xgboost#3007]
Based on the example in custom_objective.py, it looks like we cannot pass the `scale_pos_weight` as a parameter.  If my loss function requires the `scale_pos_weight`, how should I handle it?

Thanks!
